scripts/generate.jl

```diff
@@ -3,52 +3,124 @@
-const LIBXPRESS_HEADERS = [joinpath(LIBXPRESS_INCLUDE, header) for header in [
-    # "bindrv.h",
-    # "mmhttp.h",
-    # "mmquad.h",
-    # "mmsystem.h",
-    # "mmxprs.h",
-    # "mosel_dso.h",
-    # "mosel_mc.h",
-    # "mosel_ni.h",
-    # "mosel_nifct.h",
-    # "mosel_rt.h",
-    # "xprb.h",
-    # "xprb_cpp.h",
-    # "xprd.h",
-    # "xprm_mc.h",
-    # "xprm_ni.h",
-    # "xprm_rt.h",
-    # "xprnls.h",
-    "xprs.h",
-    # "xprs_mse_defaulthandler.h",
-    # "XPRSdefaultMipSolEnumHandler.java",
-    # "xslp.h",
-]]
-
-wc = init(;
-    headers = LIBXPRESS_HEADERS,
-    output_file = joinpath(@__DIR__, "../src/lib.jl"),
-    common_file = joinpath(@__DIR__, "../src/common.jl"),
-    clang_includes = vcat(LIBXPRESS_INCLUDE, CLANG_INCLUDE),
-    clang_args = ["-I", joinpath(LIBXPRESS_INCLUDE, "..")],
-    header_wrapped = (root, current) -> root == current,
-    header_library = x -> "libxprs",
-    clang_diagnostics = true,
+using Clang.Generators
+
+# Copy the appropriate xprs.h file to the scripts directory and edit these lines.
+
+old_xprs_filename = joinpath(@__DIR__, "xprs33.01.12.h")
+xprs_filename = joinpath(@__DIR__, "xprs42.01.05.h")
+
+context = create_context(
+    [xprs_filename],
+    get_default_args(),
+    load_options(joinpath(@__DIR__, "generate.toml")),
 )
+build!(context)
+
+function replace_line_cstring(line, signature)
+    changes = getindex.(
+        signature,
+        findall(r"char(\s*?\*| [a-z0-9]+?\[\])", signature),
+    )
+    last_index = 1
+    outputs = String[]
+    for (change, range) in zip(changes, findall("Ptr{UInt8}", line))
+        push!(outputs, line[last_index:(first(range)-1)])
+        if endswith(change, "*")
+            push!(outputs, "Cstring")
+        else
+            push!(outputs, line[range])
+        end
+        last_index = last(range) + 1
+    end
+    push!(outputs, line[last_index:end])
+    return join(outputs)
+end
 
-run(wc)
+function postprocess(filename)
+    contents = read(filename, String);
+    # Remove the deprecated if-else blocks
+    regex = r"if XPRSdeprecated[a-z]+\n\s+(.+?)\n\s+else\n\s+\1\s+end"s
+    while (m = match(regex, contents)) !== nothing
+        contents = replace(contents, m.match => m.captures[1])
+    end
+    # Remove mutable structs
+    regex = r"mutable struct (.+?) end"
+    while (m = match(regex, contents)) !== nothing
+        contents = replace(contents, m.match => "")
+        contents = replace(contents, m.captures[1] => "Cvoid")
+    end
+    # Remove the cenum block
+    contents = replace(contents, r"\@cenum.+?end\s+"s => "")
+    # For backwards compatibility, older versions of Xpress.jl used UInt8
+    # instead of Cchar.
+    contents = replace(contents, "Cchar" => "UInt8")
+    # For backwards compatibility, older versions of Xpress.jl used Cint
+    # instead of Clong.
+    # TODO(odow): this is wrong. Fix me in a separate PR.
+    contents = replace(contents, "Clong" => "Cint")
```

Comment:
So I found one super weird thing here. The `xprs.jl` uses `Cint` in places where it should use `Clong`. They are equivalent on windows, but not on Mac or Linux. I don't know if this causes any issues because we don't call the `64(` API.